client/src/App/hooks/use-has-time-passed.ts

```diff
@@ -8,7 +8,7 @@ import { useEffect, useRef, useState } from 'react';
  */
 export const useHasTimePassed = (duration = 500) => {
   const [hasTimePassed, setHasTimePassed] = useState(false);
-  const timeoutRef = useRef<NodeJS.Timeout>();
+  const timeoutRef = useRef<NodeJS.Timeout>(undefined);
```

Comment:
[useRef requires an argument in React 19](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#useref-requires-argument)